### PR TITLE
[10.0][ADD] medical_procedure: Medical procedures

### PR DESCRIPTION
**Pull request by obulkin**
[IMP] ``medical_emr``: Isolate ``medical.procedure``
* Move ``medical.procedure`` model, views, actions, menu item, and security rules into separate module ``medical_procedure``
* Update manifest accordingly, adding a ``medical_procedure`` dependency

[ADD] ``medical_procedure``: Isolate ``medical.procedure``
* Move ``medical.procedure`` model, views, actions, menu item, and security rules from ``medical_emr``
* Update code to use v10 syntax and align better with OCA guidelines
* Make minor enhancements to model and views

### CODE REVIEW

**Comment by lasley** [comment]
cc @jbeficent

File: medical_emr/__manifest__.py [hunk 1]
```diff
@@ -39,7 +40,9 @@
     'version': '8.0.1.1',
     'author': "Odoo Medical Team, LasLabs, Odoo Community Association (OCA)",
     'category': 'Medical',
-    'depends': ['medical'],
+    'depends': [
```
Comment (lasley) [inline medical_emr/__manifest__.py]: `medical_emr` is a deprecated module & is being stripped for parts. As such, the only changes that we should see in this module is the removal of logic that is being used elsewhere. We should not be adding anything.

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks @obulkin - procedure looks good. Just need to revert some adds in emr - explanation inline

**Comment by lasley** [comment]
It's worth noting that this is precursor to implementation of the full [FHIR Procedure](https://www.hl7.org/fhir/procedure.html) spec

**Comment by JordiBForgeFlow** [comment]
Perhaps we can find a common definition with our current WIP:

https://github.com/Eficent/vertical-medical/blob/10.0-
feature-procedurerequest/medical_procedure/README.rst

We're happy to discuss

Cc @rgarnau

**Comment by lasley** [comment]
Your version looks more complete to me IMO, we should probably just go with that. Looks like you've implemented into stuff already, and we haven't yet. 

Are you anywhere near ready for a PR on your side?

**Comment by JordiBForgeFlow** [comment]
Yes, we are planning to use intensively. We are integrating it with Care
Plans
https://github.com/Eficent/vertical-medical/tree/10.0-feature-careplan?files=1

There's plenty of stuff that we want to propose to OCA. We we've been in
experimentation mode for a while, but perhaps it is best to propose the
work now.

So far we have implemented, the following from FHIR (not strictly, though):
- Encounter
- Care Plan
- Request Group
- Procedure Request
- Procedure
- Workflow - Plan definition

The area that seems most complex is the integration with billing, and we
are not using Claims.

**Comment by lasley** [comment]
Awesome, thanks Jordi! We were about to use Procedures in the upgrade of `medical_lab`, which I believe is being formed into the [DiagnosticReport](https://www.hl7.org/fhir/diagnosticreport.html) (@obulkin correct me if I'm wrong).

Now that we have two of us working pretty heavily in this, we're diverging pretty hard. Do you think it would be beneficial to move into an OCA incubator instead of our forks? I've teased with the idea previously, but it never seemed beneficial until now.

Regarding billing - yeah that's a can of worms, even without claims. At this point in pharma, we've pretty much just been integrating billing into Odoo as much as possible - but we've been hitting a lot of walls with it, and it's a cash-only thing. 

Are you having to consider for insurances?

**Comment by obulkin** [comment]
@jbeficent - It's good to see that someone else is working on this and also trying to align with the FHIR spec. My goal here was just to represent the reusable [CodeableConcept](https://www.hl7.org/fhir/datatypes.html#CodeableConcept) portion of a procedure, the actual service that's being requested in a [ProcedureRequest](https://www.hl7.org/fhir/procedurerequest.html). Here are some examples: https://www.hl7.org/fhir/valueset-procedure-code.html. 

It looks like you've decided to handle that using the product model, which is probably a good move because of all the useful logic that already exists in Odoo for products. Let me know if that's an accurate assessment of your code as it means we can close this PR and further refactor ``medical_lab`` to use your ``medical.procedure.request`` model in place of a roughly equivalent ``medical.diagnostic.request`` model that I was planning to introduce there.

**Comment by lasley** [comment]
Closing this in favor of
* #186

**Comment by obulkin** [comment]
My understanding is that #186 has been dropped in favor of v11 development, so this PR is once again needed for bringing ``medical_diagnostic`` (formerly ``medical_lab``) to v10.

**Comment by obulkin** [comment]
``medical_emr`` has been removed from the repo due to copyright concerns, so I've refactored ``medical_procedure`` to eliminate any non-trivial use of ``medical_emr`` code. It should now be ready for a new round of review.

**Review by lasley — APPROVED** [review]
Thanks @obulkin

**Pull request merged**